swiper.el

```diff
@@ -733,53 +733,36 @@ When capture groups are present in the input, print them instead of lines."
   (when (bound-and-true-p evil-mode)
     (evil-set-jump)))
 
-(declare-function char-fold-to-regexp "char-fold")
+(defun swiper--normalize-regex (re)
+  "Normalize the swiper regex RE.
+Add a space after a leading `^' if needed and apply
+`search-default-mode' if bound."
+  (replace-regexp-in-string
+   "^\\(\\(?:\\\\(\\)?\\^\\)"
```

Comment:
Why is there a `^` at the beginning of this, rather than a
```
\\`
```
?

I would love if this were written as `(rx bol (group (? "\\(") ?^))` instead, but sadly I don't think @abo-abo likes `rx`.

Reply:
funny thing is that I actually wrote all of these regex with `rx` and just evaled and replaced the forms with strings. I would love it ivy used that library as well. Makes things so much easier to read and write.

Reply:
Thanks, was just checking because I'm not familiar with the code.